### ISSUE TITLE
UnitTestgui errors with tests on lists

### ISSUE THREAD

**Issue by aerenchyma** [RunestoneInteractive/RunestoneComponents#249]
@laumurph and I have noticed a discrepancy in test results with the unittestgui in increasing versions:

She has 2.6.2. I have the most recent components from Github (not the most recent release, the upstream/master sync). When she runs a test like:

`self.assertEqual(names, ['Paul', 'Jackie', 'Stephen'], "Testing that names has the correct values assigned")`

and names does indeed have the correct list value, the test passes. When I run it, with the same code, I get a failure: "TypeError: cannot concatenate 'str' and 'list' objects." This is only fixed, as far as I can tell, by changing the test to this:

`self.assertEqual(str(names), str(['Paul', 'Jackie', 'Stephen']), "Testing that names has the correct values assigned")`

Do you think this is related to one of the recent test updates, @bnmnetp? Has anyone else noticed something like this, and which is the ideal behavior? I imagine that testing list values should definitely be doable without casting them to strings. Any idea if I could be causing this error for any other reason? I'm using `runestone serve` to test this at the moment, which @laumurph is doing too; other tests pass as expected.

**Comment by bnmnetp** [comment]
Well, at the pace we have been going 2.6.2 is ancient.

Hmmm, I just did the following test:

```python
from unittest.gui import TestCaseGui

def foo():
    return ['1','2','3']

class myTestCase(TestCaseGui):
    def test_one(self):
        self.assertEqual(foo(), ['1','2','3'])
        

myTestCase().main()
```

which worked perfectly fine.  I am using the latest RunestoneComponents on github.

**Comment by aerenchyma** [comment]
Currently seeing the same error with the latest pip release but not with the latest development code. (If they're the same as of an hour ago, which I thought they _were_ with respect to the unittestgui,  I suspect it may be something Lauren and I are missing in the .rst files somehow, but we do see these different things with the same rst file and potentially different components installs.) At worst, something's been "fixed" on our end. At best, we'll figure out exactly what was happening tomorrow, and I will update this to explain.

**Comment by aerenchyma** [comment]
Not solved yet. The following test and code (indenting correct for me, c&p breaks it):

```

f = open("about_programming.txt") # file exists via :available_files:
program_lines = []
for l in f:
    if "program" in l:
        program_lines.append(l)
        print l

from unittest.gui import TestCaseGui

class myTests(TestCaseGui):

          def testOne(self):
             tmp = []
             for l in open("about_programming.txt").readlines():
                if "program" in l:
                   tmp.append(l)
             self.assertEqual(program_lines, tmp, "Testing that program_lines is a list of lines that contain the string 'program'")```

Gets the following:
<img width="711" alt="screen shot 2016-08-31 at 4 47 02 pm" src="https://cloud.githubusercontent.com/assets/1413839/18145784/987687ea-6f9a-11e6-88df-9fb32358e2d3.png">

It looks like the % doesn't count assertIn/assertNotIn tests, and I still get that concatenation problem. Has anyone who isn't me, @laumurph, @shdoong (in the same room here) reproduced these problems? Several people have rechecked code and rst formatting, and in some cases did not update rst files but did update components and then got new results (tests failing in this way where they did not before). Do you think it could be an ordering problem or another choice on our end that could cause this result? What concatenation occurs in the test run? I will try to take a look at this again.

**Comment by bnmnetp** [comment]
This should be fixed in runestone 2.7.4    
I was getting a similar error with a dictionary question and was finally able to track it down.  Could you verify and close?
